pkg/server/interceptors.go

```diff
@@ -43,12 +46,27 @@ func unaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServ
 	return resp, err
 }
 
-func combineInterceptors(a, b grpc.UnaryServerInterceptor) grpc.UnaryServerInterceptor {
-	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-		return a(ctx, req, info, func(actx context.Context, areq interface{}) (interface{}, error) {
-			return b(actx, areq, info, func(bctx context.Context, breq interface{}) (interface{}, error) {
-				return handler(bctx, breq)
-			})
-		})
-	}
+// the always-yes logging decider function (because grpc_zap requires a decider function)
+func yesLogIt(ctx context.Context, fullMethodName string, servingObject interface{}) bool {
+	return true
+}
+
+// WithUnaryLoggingInterceptor combines interceptors for logging all GRPC unary calls with the
+// given other unary interceptors.
+func WithUnaryLoggingInterceptor(log *zap.Logger, otherInterceptors ...grpc.UnaryServerInterceptor) grpc.UnaryServerInterceptor {
+	interceptors := append([]grpc.UnaryServerInterceptor{
+		grpc_ctxtags.UnaryServerInterceptor(),
+		grpc_zap.PayloadUnaryServerInterceptor(log, yesLogIt),
```

Comment:
i feel like we might want a this `log` to be a different zap.Logger so that these new grpc request logs can be written somewhere else and not mixed in with normal application logs. I think a new logger would be the way to do that, but not 100%, reading docs now.

Reply:
I don't actually think that will be necessary for this one-off tool. I don't expect us to use it more than once or twice, and grep should be entirely up to the task of separating the data. We should avoid spending more time than necessary creating this feature.

Reply:
if this is only going to be used "once or twice", does that mean we expect to remove this change once we are done using it?  Alternatively, if this is just a prototype, should we just keep it on its own branch and test it out there instead of merging into master?

Reply:
I would like this to be a separate branch, because there isn't any particular need for it to be in master. Stefan indicated that the changes would need to be in master before he could run it, though. I don't know whether that rule is his own or from devops. Maybe we can get an exemption?

If this does get merged in to master, I don't think it's vital that it stay there. It could come in handy for developers in some debugging scenarios, but it's reasonable to ask developers to apply a temporary patch if they need this functionality.

Reply:
hmm tricky. i guess in that case sounds like it might be easier to merge to master, even though its less than ideal.